scripts/file_resume.js

```diff
@@ -48,9 +48,15 @@ $(function() {
 
     // Before we start the upload, prevent the user from hitting upload again.
     $("#old_submit").click(function(ev) {
-        let filename = $("#old_browse")[0].files[0].name;
-        if(validate(filename)) {
-            // won't work if we disable browse button
+        let file = $("#old_browse")[0].files[0];
+        if(file.size >= maxNormSize) {
```

Comment:
I don't understand this change. If the user has Javascript enabled they will very likely be using the resumable uploader. If they don't, this code will never run and this check will never happen. I don't think it's worth supporting the very small set of users who will have JS enabled but won't support resumable uploads.

Reply:
This was to avoid the difficult situation that happens when a too-big file is uploaded and is handled by the function 'detect_too_large()' on line 23 of upload_file.inc. But as you say it is unlikely to happen. Sharon's testing came across it but only because I hacked the code to disable resumable in order to test the non-resumable case.